### PR TITLE
 API1600 support for Interconnects

### PR DESCRIPTION
**Pull request by chebroluharika**
### Description
API1600 support for Interconnects

### Issues Resolved
NA

### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass (`$ rake test`).
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [x] New endpoints supported are updated in the endpoints-support.md file.
- [x] Changes are documented in the CHANGELOG.

### CODE REVIEW

**Comment by soodpr** [comment]
Please raise LI resource PR as well as Interconnect has dependency on LI. I will review this later

File: examples/shared_samples/interconnect.rb [hunk 1]
```diff
@@ -15,16 +15,10 @@
...
-# api_version = 200 & variant = any to OneviewSDK::API200::Interconnect'
-# api_version = 300 & variant = C7000 to OneviewSDK::API300::C7000::Interconnect'
-# api_version = 300 & variant = Synergy to OneviewSDK::API300::Synergy::Interconnect'
-# api_version = 500 & variant = C7000 to OneviewSDK::API500::C7000::Interconnect'
-# api_version = 500 & variant = Synergy to OneviewSDK::API500::Synergy::Interconnect'
-# api_version = 600 & variant = C7000 to OneviewSDK::API600::C7000::Interconnect'
-# api_version = 600 & variant = Synergy to OneviewSDK::API600::Synergy::Interconnect'
+# - 200, 300, 500, 600, 800, 1000, 1200, 1600
+# Supoported Variants
+# C7000, Synergy
```
Comment (AsisBagga) [inline examples/shared_samples/interconnect.rb:20]: Not sure why parameter details were removed. Need to add them.

File: lib/oneview-sdk/resource/api1600/c7000/interconnect.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      class Interconnect < OneviewSDK::API1200::C7000::Interconnect
+        include OneviewSDK::ResourceHelper::ConfigurationOperation
+        # Create a resource object, associate it with a client, and set its properties.
+        # @param [OneviewSDK::Client] client The client object for the OneView appliance
+        # @param [Hash] params The options for this resource (key-value pairs)
+        # @param [Integer] api_ver The api version to use when interracting with this resource.
+        def initialize(client, params = {}, api_ver = nil)
+          @data ||= {}
+          # Default values
+          @data['type'] ||= 'InterconnectV7'
```
Comment (AsisBagga) [inline lib/oneview-sdk/resource/api1600/c7000/interconnect.rb:27]: There is no interconnectV7, please remove the initialize method if same method exist in API1200.

File: lib/oneview-sdk/resource/api1600/synergy/interconnect.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      # Interconnect resource implementation on API1600 Synergy
+      class Interconnect < OneviewSDK::API1200::Synergy::Interconnect
+        # Create a resource object, associate it with a client, and set its properties.
+        # @param [OneviewSDK::Client] client The client object for the OneView appliance
+        # @param [Hash] params The options for this resource (key-value pairs)
+        # @param [Integer] api_ver The api version to use when interracting with this resource.
+        def initialize(client, params = {}, api_ver = nil)
+          @data ||= {}
+          # Default values
+          @data['type'] ||= 'InterconnectV7'
```
Comment (AsisBagga) [inline lib/oneview-sdk/resource/api1600/synergy/interconnect.rb:26]: same as above.

File: spec/unit/resource/api1600/c7000/interconnect_spec.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  include_context 'shared context'
+
+  it 'inherits from OneviewSDK::API1200::C7000::Interconnect' do
+    expect(described_class).to be < OneviewSDK::API1200::C7000::Interconnect
+  end
+
+  describe '#initialize' do
+    it 'should be initialize the instance with default values' do
+      item = described_class.new(@client_1600)
+      expect(item['type']).to eq('InterconnectV7')
```
Comment (AsisBagga) [inline spec/unit/resource/api1600/c7000/interconnect_spec.rb:13]: please correct this as well

File: spec/unit/resource/api1600/synergy/interconnect_spec.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  include_context 'shared context'
+
+  it 'inherits from OneviewSDK::API1200::Synergy::Interconnect' do
+    expect(described_class).to be < OneviewSDK::API1200::Synergy::Interconnect
+  end
+
+  describe '#initialize' do
+    it 'should be initialize the instance with default values' do
+      item = described_class.new(@client_1600)
+      expect(item['type']).to eq('InterconnectV7')
```
Comment (AsisBagga) [inline spec/unit/resource/api1600/synergy/interconnect_spec.rb:13]: same as above

File: examples/shared_samples/interconnect.rb (hunk 1 above)
Comment (chebroluharika) [reply examples/shared_samples/interconnect.rb:20]: done

File: lib/oneview-sdk/resource/api1600/c7000/interconnect.rb (hunk 2 above)
Comment (chebroluharika) [reply lib/oneview-sdk/resource/api1600/c7000/interconnect.rb]: done

File: lib/oneview-sdk/resource/api1600/synergy/interconnect.rb (hunk 3 above)
Comment (chebroluharika) [reply lib/oneview-sdk/resource/api1600/synergy/interconnect.rb]: done

File: spec/unit/resource/api1600/c7000/interconnect_spec.rb (hunk 4 above)
Comment (chebroluharika) [reply spec/unit/resource/api1600/c7000/interconnect_spec.rb]: done

File: spec/unit/resource/api1600/synergy/interconnect_spec.rb (hunk 5 above)
Comment (chebroluharika) [reply spec/unit/resource/api1600/synergy/interconnect_spec.rb]: done

**Pull request merged**